restli-tools/src/main/java/com/linkedin/restli/tools/data/ExtensionSchemaValidationCmdLineApp.java

```diff
@@ -41,11 +50,24 @@
 import org.slf4j.LoggerFactory;
 
 
+/**
+ * This class is used to validate extension schema, the validation covers following parts:
+ * 1. The extension schema is a validate schema.
+ * 2. The extension schema's fields should only be annotated by "extension" namespace.
+ * 3. The extension schema's fields annotations should be {@link ExtensionSchemaAnnotation}.
+ * 4. The extension schema's field's schema should only be typeRef or array of typeRef.
```

Comment:
Why only typerefs? Wouldn't checking for resourceKey annotation be enough?

Reply:
Based on LinkedIn TypedUrn concept, check resourceKey annotation should be good enough. I will update the logic here.